### PR TITLE
fix: allow leading digits in scope and package names

### PR DESCRIPTION
**Pull request by crowlKats**

### CODE REVIEW

**Review by marvinhagemeister — APPROVED** [review]

**Review by lucacasonato — COMMENTED** [review]
Does deno_doc already understand it can't use leading digits in identifiers in usage blocks?

**Comment by crowlKats** [comment]
> Does deno_doc already understand it can't use leading digits in identifiers in usage blocks?

I am not sure I understand the question

**Comment by marvinhagemeister** [comment]
@crowlKats It's about the usage block:

```ts
// 123lib is not a valid JS identifier. JS identifiers must not start with a number
import 123lib from "@foo/123lib";
```

Might be that this is already handled in `deno_doc`

**Comment by crowlKats** [comment]
oh i see, its about auto detection of default import names. hm, that is not handled currently

File: api/src/npm/tarball.rs [hunk 1]
```diff
@@ -133,7 +133,13 @@ pub async fn create_npm_tarball<'a>(
...
             .expect("parsed source should be here");
 
-          let module_info = DefaultModuleAnalyzer::module_info(&source);
+          let analyzer = DefaultModuleAnalyzer;
+
+          let module_info = analyzer.analyze(
+            module.specifier(),
+            source.text_info().text(),
+            js.media_type,
+          )?;
```
Comment (dsherret) [inline api/src/npm/tarball.rs:142]: This will reparse. I think it's now: `ParsedModuleAnalyzer::modulr_info(&source)`

**Review by dsherret — APPROVED** [review]
LGTM

**Comment by crowlKats** [comment]
@lucacasonato i just looked through the docs and didnt find anything that talks about the first character/requiring a starting letter

**Comment by lucacasonato** [comment]
`scopes.md` and `packages.md` both mention it.

**Comment by crowlKats** [comment]
@lucacasonato i guess i really cannot read; fixed it up now

**Review by marvinhagemeister — APPROVED** [review]

**Comment by lucacasonato** [comment]
Great, TY!

**Pull request merged**